astropy/units/physical.py

```diff
@@ -311,6 +313,19 @@ def __init__(self, unit, physical_types):
     def __iter__(self):
         yield from self._physical_type_list
 
+    def __getattr__(self, attr):
+        if attr in dir(str):
+            warning_message = (
+                f"support for accessing str attributes such as {attr} "
+                "from PhysicalType instances is deprecated since 4.3 and will be removed in a "
+                "subsequent release.")
+            warnings.warn(warning_message, AstropyDeprecationWarning)
+            self_str_attr = getattr(str(self), attr, None)
```

Comment:
No need for the ", None" any more if `attr in dir(str)`, right?

Note that personally I'd prefer to stick with keeping this on top - if we enter `__getattr__`, regular attribute access has failed already so it doesn't really matter if it is slightly slower. And I'm just not 100% sure that `dir(str)` and `hasattr(str(self), attr)` are really completely equivalent...